src/DGtal/shapes/parametric/Astroid2D.ih

```diff
@@ -49,19 +56,19 @@ DGtal::Astroid2D<T>::~Astroid2D()
 template <typename T>
 inline
 DGtal::Astroid2D<T>::Astroid2D( const double x0, const double y0,
-  const double a, const double b ) : myCenter(x0,y0), myA(a), myB(b)
+  const double a, const double b ) : myCenter(x0,y0), myA(fabs(a)), myB(fabs(b))
```

Comment:
I don't know if the best practice with DGtal is that or to throw an Exception in case of negative value.
What do you think about @copyme?

Reply:
Well, for me `fabs` look ok, as long this behavior is documented.

Reply:
Perfect, I still added a comment on .h file.